### ISSUE TITLE
greedyWrap doesn't wrap paragraphs correctly

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5144]
|<img alt="termim's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| termim reported|
|-|-|
|Trac ID|trac#5144|
|Type|defect|
|Created|2011-06-08 03:24:03Z|

**greedyWrap** doesn't pass **width** parameter recursively when wrapping paragraphs.

Attachments:

* [text.wrap.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/042/042a444b55411bb7c8c71613cfa517c4ca05744e/90cc307c86b4ef8687fdeea463983b575ddd1400.diff) (1207 bytes) - added by termim on 2011-06-08 03:25:09Z - fix for twisted.python.text.greedyWrap to wrap paragraphs correctly

<details><summary>Searchable metadata</summary>

```
trac-id__5144 5144
type__defect defect
reporter__termim termim
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__usage_wrap usage wrap
time__1307503443000000 1307503443000000
changetime__1307574705000000 1307574705000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @exarkun|
|-|-|

Extra points for the subtle test change. :)

Thanks, this looks good.  I'll apply it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [32074]) Apply patch fixing greedyWrap's line wrapping of paragraphs

Author: termim
Reviewer: exarkun
Fixes: [#5144](https://github.com/twisted/twisted/issues/5144)

In `greedyWrap`, pass along the width value to recursive calls used to wrap
paragraphs within the text, so that each paragraph is wrapped to the desired
width.
